### PR TITLE
Add aws_quicksight_group resource

### PR DESCRIPTION
**Pull request by kterada0509**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Reference #8146 

Changes proposed in this pull request:

* Add `aws_quicksight_group` resource

Output from acceptance testing:

```
$  make testacc TEST=./aws TESTARGS='-run=TestAccAWSQuickSightGroup_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -parallel 20 -run=TestAccAWSQuickSightGroup_ -timeout 120m
=== RUN   TestAccAWSQuickSightGroup_basic
=== PAUSE TestAccAWSQuickSightGroup_basic
=== RUN   TestAccAWSQuickSightGroup_withDescription
=== PAUSE TestAccAWSQuickSightGroup_withDescription
=== RUN   TestAccAWSQuickSightGroup_disappears
=== PAUSE TestAccAWSQuickSightGroup_disappears
=== CONT  TestAccAWSQuickSightGroup_basic
=== CONT  TestAccAWSQuickSightGroup_withDescription
=== CONT  TestAccAWSQuickSightGroup_disappears
--- PASS: TestAccAWSQuickSightGroup_disappears (35.70s)
--- PASS: TestAccAWSQuickSightGroup_withDescription (55.69s)
--- PASS: TestAccAWSQuickSightGroup_basic (56.63s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	56.694s
```

### CODE REVIEW

**Comment by kterada0509** [comment]
@bflad 
Could you review this pr ?

File: aws/resource_aws_quicksight_group.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+		}
+		return fmt.Errorf("Error deleting Quick Sight Group %s: %s", d.Id(), err)
+	}
+
+	return nil
+}
+
+func resourceAwsQuickSightGroupParseID(id string) (string, string, string, error) {
+	parts := strings.SplitN(id, "/", 3)
+	if parts[0] == "" || parts[1] == "" || parts[2] == "" {
```
Comment (bflad) [inline aws/resource_aws_quicksight_group.go]:
```suggestion
	if len(parts) < 3 || parts[0] == "" || parts[1] == "" || parts[2] == "" {
```

File: aws/resource_aws_quicksight_group_test.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+			GroupName:    aws.String(groupName),
+		}
+
+		output, err := conn.DescribeGroup(input)
+
+		if err != nil {
+			return err
+		}
+
+		if output.Group == nil {
```
Comment (bflad) [inline aws/resource_aws_quicksight_group_test.go]:
```suggestion
		if output == nil || output.Group == nil {
```

File: aws/resource_aws_quicksight_group_test.go [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+		if err != nil {
+			return err
+		}
+
+		_, err = client.DescribeGroup(&quicksight.DescribeGroupInput{
+			AwsAccountId: aws.String(awsAccountID),
+			Namespace:    aws.String(namespace),
+			GroupName:    aws.String(groupName),
+		})
+		if !isAWSErr(err, quicksight.ErrCodeResourceNotFoundException, "") {
```
Comment (bflad) [inline aws/resource_aws_quicksight_group_test.go]: We generally prefer to split this out so we can display any API errors 👍 

```go
if isAWSErr(err, quicksight.ErrCodeResourceNotFoundException, "") {
	continue
}

if err != nil {
	return err
}

return fmt.Errorf("Quick Sight Group '%s' was not deleted properly", rs.Primary.ID)
```

File: aws/resource_aws_quicksight_group_test.go [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+			return err
+		}
+
+		return nil
+	}
+}
+
+func testAccAWSQuickSightGroupConfig(rName string) string {
+	return fmt.Sprintf(`
+provider "aws" {
```
Comment (bflad) [inline aws/resource_aws_quicksight_group_test.go]: For now, please omit hardcoded `provider` configurations in test configurations and rely on the environment variable handling. We'll eventually be circling around to fix the problems caused by this type of testing setup. 😄 See also: #8316 and #8983

File: aws/resource_aws_quicksight_group_test.go [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+	region = "us-east-1"
+}
+
+data "aws_caller_identity" "current" {}
+
+resource "aws_quicksight_group" "default" {
+	aws_account_id 	= "${data.aws_caller_identity.current.account_id}"
+	group_name 		= "%[1]s"
```
Comment (bflad) [inline aws/resource_aws_quicksight_group_test.go]: Nit: formatting seems a little off, tabs versus spaces? 😎

File: aws/resource_aws_quicksight_group_test.go [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+}
+
+`, rName)
+
+}
+
+func testAccAWSQuickSightGroupConfigWithDescription(rName, description string) string {
+	return fmt.Sprintf(`
+provider "aws" {
```
Comment (bflad) [inline aws/resource_aws_quicksight_group_test.go]: Same here re: provider configuration 👍

File: aws/resource_aws_quicksight_group.go [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"aws_account_id": {
+				Type:     schema.TypeString,
+				Required: true,
```
Comment (bflad) [inline aws/resource_aws_quicksight_group.go]: This argument can be made optional by reading the AWS Account ID information from the provider itself, e.g.

```go
	awsAccountID := meta.(*AWSClient).accountid
	namespace := d.Get("namespace").(string)

	if v, ok := d.GetOk("aws_account_id"); ok {
		awsAccountID = v.(string)
	}

	createOpts := &quicksight.CreateGroupInput{
		AwsAccountId: aws.String(awsAccountID),
		Namespace:    aws.String(namespace),
		GroupName:    aws.String(d.Get("group_name").(string)),
	}
```

Then the testing can/should omit the `aws_caller_identity` data source. 👍 

```suggestion
				Optional: true,
```

File: website/docs/r/quicksight_group.html.markdown [hunk 8]
```diff
@@ -0,0 +1,44 @@
+---
+layout: "aws"
+page_title: "AWS: aws_quicksight_group"
+sidebar_current: "docs-aws-resource-quicksight-group"
+description: |-
+  Manages a Resource Quick Sight Group.
+---
+
+# aws_quicksight_group
```
Comment (bflad) [inline website/docs/r/quicksight_group.html.markdown]: We likely added this to the other resource pages after this PR was submitted

```suggestion
# Resource: aws_quicksight_group
```

File: website/docs/r/quicksight_group.html.markdown [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+Resource for managing Quick Sight Group
+
+## Example Usage
+
+```hcl
+resource "aws_quicksight_group" "example" {
+	aws_account_id 	= "123456789123"
```
Comment (bflad) [inline website/docs/r/quicksight_group.html.markdown]: When this argument is made optional, let's omit this here as well. 👍

File: website/docs/r/quicksight_group.html.markdown [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+	group_name 		= "tf-example"
```
Comment (bflad) [inline website/docs/r/quicksight_group.html.markdown]: Nit: formatting

```suggestion
  group_name = "tf-example"
```

File: website/docs/r/quicksight_group.html.markdown [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `aws_account_id` - (Required) The ID for the AWS account that the group is in. Currently, you use the ID for the AWS account that contains your Amazon QuickSight account.
+* `group_name` - (Required) A name for the group.
+  given by `source_ami_region`.
```
Comment (bflad) [inline website/docs/r/quicksight_group.html.markdown]: This looks extraneous 😉 

```suggestion
```

**Comment by kterada0509** [comment]
Re-run acctest

```
ture/add-support-for-aws_quicksight_group-resource *)$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSQuickSightGroup_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -parallel 20 -run=TestAccAWSQuickSightGroup_ -timeout 120m
=== RUN   TestAccAWSQuickSightGroup_basic
=== PAUSE TestAccAWSQuickSightGroup_basic
=== RUN   TestAccAWSQuickSightGroup_withDescription
=== PAUSE TestAccAWSQuickSightGroup_withDescription
=== RUN   TestAccAWSQuickSightGroup_disappears
=== PAUSE TestAccAWSQuickSightGroup_disappears
=== CONT  TestAccAWSQuickSightGroup_basic
=== CONT  TestAccAWSQuickSightGroup_disappears
=== CONT  TestAccAWSQuickSightGroup_withDescription
--- PASS: TestAccAWSQuickSightGroup_disappears (28.91s)
--- PASS: TestAccAWSQuickSightGroup_basic (65.12s)
--- PASS: TestAccAWSQuickSightGroup_withDescription (69.85s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	69.965s
```

File: website/docs/r/quicksight_group.html.markdown (hunk 11 above)
Comment (kterada0509) [reply website/docs/r/quicksight_group.html.markdown]: removed. 😅

File: website/docs/r/quicksight_group.html.markdown (hunk 9 above)
Comment (kterada0509) [reply website/docs/r/quicksight_group.html.markdown]: fixed.

File: website/docs/r/quicksight_group.html.markdown (hunk 10 above)
Comment (kterada0509) [reply website/docs/r/quicksight_group.html.markdown]: fixed.

File: website/docs/r/quicksight_group.html.markdown (hunk 8 above)
Comment (kterada0509) [reply website/docs/r/quicksight_group.html.markdown]: fixed

File: aws/resource_aws_quicksight_group.go (hunk 7 above)
Comment (kterada0509) [reply aws/resource_aws_quicksight_group.go]: fixed.

File: aws/resource_aws_quicksight_group_test.go (hunk 4 above)
Comment (kterada0509) [reply aws/resource_aws_quicksight_group_test.go]: removed.

File: aws/resource_aws_quicksight_group_test.go (hunk 3 above)
Comment (kterada0509) [reply aws/resource_aws_quicksight_group_test.go]: fixed.

File: aws/resource_aws_quicksight_group_test.go (hunk 2 above)
Comment (kterada0509) [reply aws/resource_aws_quicksight_group_test.go]: fixed.

File: aws/resource_aws_quicksight_group.go (hunk 1 above)
Comment (kterada0509) [reply aws/resource_aws_quicksight_group.go]: fixed.

File: aws/resource_aws_quicksight_group.go [hunk 12]
```diff
@@ -0,0 +1,181 @@
...
+			"group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"namespace": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "default",
```
[the remark this answers is not in the document]
Comment (kterada0509) [reply aws/resource_aws_quicksight_group.go]: fixed.

**Comment by nywilken** [comment]
The new `aws_quicksight_group` resource, has been released in [version 2.20.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2200-july-19-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

For further feature requests or bug reports with this resource, please create a [new GitHub issue](https://github.com/terraform-providers/terraform-provider-aws/issues/new/choose) following the template for triage. Thanks!

**Pull request merged**